### DISCUSSION TITLE
how stop bluetooth

### DISCUSSION THREAD

**Discussion by aqildad-create** [pschatzmann/arduino-audio-tools#592] [Q&A] [answered]
sir , bluetooth rsv audio is working but later at some point  need to connect esp with with wifi so before connecting wifi how to off the blue tooth ?

**Comment by pschatzmann** [comment, marked as answer]
The best way is to restart the ESP32 (e.g. with ESP.restart()). 
This is the only way to make sure that it is in a clean state. Just closing A2DP does not release all memory or it leaves it in a state where A2DP can not be started any more...

**Reply by AllanTee** [reply]
Switching from BT (with ESP.restart) to Wifi only succeeds on the first try. 
Presumably SW_CPU_RESET doesn't work in that situation. After POWERON_RESET it works.
Please see the logs below.

I also tried the opposite case - initially the Wifi is started, then programmatically switched to BT receiver. I have managed to close the Audio (player.end(); decoder.end(); analog.end();) but without success, it crashes.

Please give a hint, how to switch from audio tools to A2DP or vice versa.


 *  Executing task: C:\Users\A\.platformio\penv\Scripts\platformio.exe run --target upload --target monitor --environment esp32dev 

Processing esp32dev (platform: espressif32; board: esp32dev; framework: arduino)
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------!!! WARNING !!!         !!! WARNING !!!         !!! WARNING !!!
Your project is located on a mapped network drive but the current command-line shell does not support the UNC paths.
Please move your project to a physical drive or check this workaround: https://bit.ly/3kuU5mP

Verbose mode can be enabled via `-v, --verbose` option
CONFIGURATION: https://docs.platformio.org/page/boards/espressif32/esp32dev.html
PLATFORM: Espressif 32 (6.0.1+sha.764527e) > Espressif ESP32 Dev Module
HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash
DEBUG: Current (cmsis-dap) External (cmsis-dap, esp-bridge, esp-prog, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)
PACKAGES:
 - framework-arduinoespressif32 @ 3.20006.221224 (2.0.6)
 - tool-esptoolpy @ 1.40400.0 (4.4.0)
 - tool-mkfatfs @ 2.0.1
 - tool-mklittlefs @ 1.203.210628 (2.3)
 - tool-mkspiffs @ 2.230.0 (2.30)
 - toolchain-xtensa-esp32 @ 8.4.0+2021r2-patch5
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ soft
Found 37 compatible libraries
Scanning dependencies...
Dependency Graph
|-- ESP32-A2DP @ 1.7.1+sha.07550a3  
|-- audio-tools @ 0.9.3+sha.4c8820a 
|   |-- libhelix @ 0.1.0+sha.b706f34
|   |-- WiFiClientSecure @ 2.0.0    
|   |   |-- WiFi @ 2.0.0
|   |-- WiFi @ 2.0.0
|-- libhelix @ 0.1.0+sha.b706f34
|-- EasyButton @ 2.0.1
|-- LittleFS @ 2.0.0
|   |-- FS @ 2.0.0
Building in release mode
Retrieving maximum program size Z:\Dokumendid\IT\PlatformIO\Projects\Speaker\Player_0.4.2\.pio\build\esp32dev\firmware.elf
Checking size Z:\Dokumendid\IT\PlatformIO\Projects\Speaker\Player_0.4.2\.pio\build\esp32dev\firmware.elf
Advanced Memory Usage is available via "PlatformIO Home > Project Inspect"
RAM:   [==        ]  18.5% (used 60764 bytes from 327680 bytes)   
Flash: [========  ]  83.1% (used 1742605 bytes from 2097152 bytes)
Configuring upload protocol...
AVAILABLE: cmsis-dap, esp-bridge, esp-prog, espota, esptool, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa
CURRENT: upload_protocol = esptool
Looking for upload port...
Auto-detected: COM3
Uploading Z:\Dokumendid\IT\PlatformIO\Projects\Speaker\Player_0.4.2\.pio\build\esp32dev\firmware.bin
esptool.py v4.4
Serial port COM3
Connecting.....
Chip is ESP32-D0WDQ6 (revision v1.0)
Features: WiFi, BT, Dual Core, 240MHz, VRef calibration in efuse, Coding Scheme None
Crystal is 40MHz
MAC: 9c:9c:1f:c9:70:34
Uploading stub...
Running stub...
Stub running...
Changing baud rate to 460800
Changed.
Configuring flash size...
Flash will be erased from 0x00001000 to 0x00005fff...
Flash will be erased from 0x00008000 to 0x00008fff...
Flash will be erased from 0x0000e000 to 0x0000ffff...
Flash will be erased from 0x00010000 to 0x001bafff...
Compressed 17408 bytes to 12112...
Writing at 0x00001000... (100 %)
Wrote 17408 bytes (12112 compressed) at 0x00001000 in 0.6 seconds (effective 218.3 kbit/s)...
Hash of data verified.
Compressed 3072 bytes to 134...
Writing at 0x00008000... (100 %)
Wrote 3072 bytes (134 compressed) at 0x00008000 in 0.1 seconds (effective 312.5 kbit/s)...
Hash of data verified.
Compressed 8192 bytes to 47...
Writing at 0x0000e000... (100 %)
Wrote 8192 bytes (47 compressed) at 0x0000e000 in 0.1 seconds (effective 446.8 kbit/s)...
Hash of data verified.
Compressed 1748384 bytes to 1129147...
Writing at 0x00010000... (1 %)
Writing at 0x0001b5e5... (2 %)
Writing at 0x000275a9... (4 %)
...
...
...
Writing at 0x001aa5be... (97 %)
Writing at 0x001b044a... (98 %)
Writing at 0x001b585a... (100 %)
Wrote 1748384 bytes (1129147 compressed) at 0x00010000 in 26.5 seconds (effective 527.2 kbit/s)...
Hash of data verified.

Leaving...
Hard resetting via RTS pin...
======================================================================================================= [SUCCESS] Took 46.28 seconds =======================================================================================================

Please build project in debug configuration to get more details about an exception.
See https://docs.platformio.org/page/projectconf/build_configurations.html


--- Terminal on COM3 | 115200 8-N-1
--- Available filters and text transformations: colorize, debug, default, direct, esp32_exception_decoder, hexlify, log2file, nocontrol, printable, send_on_enter, time
--- More details at https://bit.ly/pio-monitor-filters
--- Quit: Ctrl+C | Menu: Ctrl+T | Help: Ctrl+T followed by Ctrl+H

=== Starting BT receiver.
=== btModeIsActive=0
=== Starting UrlRadio.
[I] AnalogAudioBase.h : 57 - I2S_MODE_DAC_BUILT_IN
[I] AnalogAudio.h : 57 - bool audio_tools::AnalogAudioStream::begin(audio_tools::AnalogConfig)
[I] AnalogAudioESP32.h : 48 - virtual bool audio_tools::AnalogDriverESP32::begin(audio_tools::AnalogConfig)
[I] AudioTypes.h : 63 - sample_rate: 44100
[I] AudioTypes.h : 64 - channels: 2
[I] AudioTypes.h : 65 - bits_per_sample: 16
[I] AnalogAudioBase.h : 72 - analog left output pin: 25
[I] AnalogAudioBase.h : 73 - analog right output pin: 26
[I] AnalogAudioESP32.h : 106 - TX_MODE
[I] MetaDataID3.h : 564 - virtual void audio_tools::MetaDataID3::begin()
[I] AudioSource.h : 175 - selectStream: 0/2 -> http://stream.srg-ssr.ch/m/rsj/mp3_128
[I] URLStream.h : 76 - virtual bool audio_tools::URLStream::begin(const char*, const char*, MethodID, const char*, const char*): http://stream.srg-ssr.ch/m/rsj/mp3_128
[I] Url.h : 74 - Url::parse
[I] Url.h : 110 - url->http://stream.srg-ssr.ch/m/rsj/mp3_128
[I] Url.h : 111 - host->stream.srg-ssr.ch
[I] Url.h : 112 - protocol->http
[I] Url.h : 113 - path->/m/rsj/mp3_128
[I] Url.h : 114 - port->80
.
[I] URLStream.h : 256 - WiFiClient
[I] HttpRequest.h : 193 - process connecting to host stream.srg-ssr.ch port 80
[I] HttpRequest.h : 181 - connected 1 timeout 60000
[I] HttpRequest.h : 204 - Free heap: 100876
[I] HttpHeader.h : 358 - HttpRequestHeader::setValues - path: /m/rsj/mp3_128
[I] HttpHeader.h : 255 - HttpHeader::write
[I] HttpHeader.h : 367 - HttpRequestHeader::write1stLine
[I] HttpHeader.h : 382 - -> GET /m/rsj/mp3_128 HTTP/1.1
[I] HttpHeader.h : 192 -  -> Host: stream.srg-ssr.ch 
[I] HttpHeader.h : 192 -  -> Connection: close
[I] HttpHeader.h : 192 -  -> Accept: audio/mp3 
[I] HttpHeader.h : 298 -  -> <CR LF>
[I] HttpRequest.h : 227 - Request written ... waiting for reply
[I] HttpHeader.h : 225 - HttpHeader::read
[W] HttpHeader.h : 232 - Waiting for data...
[I] HttpHeader.h : 162 - HttpHeader::readLine -> HTTP/1.1 200 OK
[I] HttpHeader.h : 162 - HttpHeader::readLine -> Content-Type: audio/mpeg
[I] HttpHeader.h : 162 - HttpHeader::readLine -> Date: Mon, 20 Feb 2023 09:37:34 GMT
[I] HttpHeader.h : 162 - HttpHeader::readLine -> icy-br:128
[I] HttpHeader.h : 162 - HttpHeader::readLine -> ice-audio-info: bitrate=128
[I] HttpHeader.h : 162 - HttpHeader::readLine -> icy-br:128
[I] HttpHeader.h : 162 - HttpHeader::readLine -> icy-description:SwissJazz_MP3@128Kbps
[I] HttpHeader.h : 162 - HttpHeader::readLine -> icy-genre:Entertainment
[I] HttpHeader.h : 162 - HttpHeader::readLine -> icy-mimetype:audio/mpeg
[I] HttpHeader.h : 162 - HttpHeader::readLine -> icy-name:/SwissJazz_Room2
[I] HttpHeader.h : 162 - HttpHeader::readLine -> icy-pub:0
[I] HttpHeader.h : 162 - HttpHeader::readLine -> icy-url:www.srgssr.ch
[I] HttpHeader.h : 162 - HttpHeader::readLine -> Server: Icecast 2.4.0-kh10
[I] HttpHeader.h : 162 - HttpHeader::readLine -> Cache-Control: no-cache, no-store
[I] HttpHeader.h : 162 - HttpHeader::readLine -> Access-Control-Allow-Origin: *
[I] HttpHeader.h : 162 - HttpHeader::readLine -> Access-Control-Allow-Headers: Origin, Accept, X-Requested-With, Content-Type
[I] HttpHeader.h : 162 - HttpHeader::readLine -> Access-Control-Allow-Methods: GET, OPTIONS, HEAD
[I] HttpHeader.h : 162 - HttpHeader::readLine -> Connection: Close
[I] HttpHeader.h : 162 - HttpHeader::readLine -> Expires: Mon, 26 Jul 1997 05:00:00 GMT
[I] HttpHeader.h : 162 - HttpHeader::readLine -> 
[W] HttpRequest.h : 153 - no CONTENT_LENGTH found in reply
[I] URLStream.h : 91 - size: 0
[I] AudioCopy.h : 69 - buffer_size=1024
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioPlayer.h : 218 - sample_rate: 48000
[I] AudioPlayer.h : 219 - bits_per_sample: 16
[I] AudioPlayer.h : 220 - channels: 2
[I] VolumeStream.h : 183 - setVolume: 1.000000
[I] VolumeStream.h : 183 - setVolume: 1.000000
[I] AudioTypes.h : 63 - sample_rate: 48000
[I] AudioTypes.h : 64 - channels: 2
[I] AudioTypes.h : 65 - bits_per_sample: 16
[I] AnalogAudio.h : 37 - virtual void audio_tools::AnalogAudioStream::setAudioInfo(audio_tools::AudioBaseInfo)
[I] AudioTypes.h : 63 - sample_rate: 48000
[I] AudioTypes.h : 64 - channels: 2
[I] AudioTypes.h : 65 - bits_per_sample: 16
[I] AnalogAudioBase.h : 72 - analog left output pin: 25
[I] AnalogAudioBase.h : 73 - analog right output pin: 26
[I] AnalogAudio.h : 63 - virtual void audio_tools::AnalogAudioStream::end()
[I] AnalogAudioESP32.h : 132 - end
[I] AnalogAudio.h : 57 - bool audio_tools::AnalogAudioStream::begin(audio_tools::AnalogConfig)
[I] AnalogAudioESP32.h : 48 - virtual bool audio_tools::AnalogDriverESP32::begin(audio_tools::AnalogConfig)
[I] AudioTypes.h : 63 - sample_rate: 48000
[I] AudioTypes.h : 64 - channels: 2
[I] AudioTypes.h : 65 - bits_per_sample: 16
[I] AnalogAudioBase.h : 72 - analog left output pin: 25
[I] AnalogAudioBase.h : 73 - analog right output pin: 26
[I] AnalogAudioESP32.h : 106 - TX_MODE
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
=== btModeIsActive=1
=== Restarting...
ets Jun  8 2016 00:22:57

rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:1184
load:0x40078000,len:13104
load:0x40080400,len:3036
entry 0x400805e4

=== Starting BT receiver.
=== btModeIsActive=0
=== Starting UrlRadio.
[I] AnalogAudioBase.h : 57 - I2S_MODE_DAC_BUILT_IN
[I] AnalogAudio.h : 57 - bool audio_tools::AnalogAudioStream::begin(audio_tools::AnalogConfig)
[I] AnalogAudioESP32.h : 48 - virtual bool audio_tools::AnalogDriverESP32::begin(audio_tools::AnalogConfig)
[I] AudioTypes.h : 63 - sample_rate: 44100
[I] AudioTypes.h : 64 - channels: 2
[I] AudioTypes.h : 65 - bits_per_sample: 16
[I] AnalogAudioBase.h : 72 - analog left output pin: 25
[I] AnalogAudioBase.h : 73 - analog right output pin: 26
[I] AnalogAudioESP32.h : 106 - TX_MODE
[I] MetaDataID3.h : 564 - virtual void audio_tools::MetaDataID3::begin()
[I] AudioSource.h : 175 - selectStream: 0/2 -> http://stream.srg-ssr.ch/m/rsj/mp3_128
[I] URLStream.h : 76 - virtual bool audio_tools::URLStream::begin(const char*, const char*, MethodID, const char*, const char*): http://stream.srg-ssr.ch/m/rsj/mp3_128
[I] Url.h : 74 - Url::parse
[I] Url.h : 110 - url->http://stream.srg-ssr.ch/m/rsj/mp3_128
[I] Url.h : 111 - host->stream.srg-ssr.ch
[I] Url.h : 112 - protocol->http
[I] Url.h : 113 - path->/m/rsj/mp3_128
[I] Url.h : 114 - port->80
.
[I] URLStream.h : 256 - WiFiClient
[I] HttpRequest.h : 193 - process connecting to host stream.srg-ssr.ch port 80
[I] HttpRequest.h : 181 - connected 1 timeout 60000
[I] HttpRequest.h : 204 - Free heap: 100864
[I] HttpHeader.h : 358 - HttpRequestHeader::setValues - path: /m/rsj/mp3_128
[I] HttpHeader.h : 255 - HttpHeader::write
[I] HttpHeader.h : 367 - HttpRequestHeader::write1stLine
[I] HttpHeader.h : 382 - -> GET /m/rsj/mp3_128 HTTP/1.1
[I] HttpHeader.h : 192 -  -> Host: stream.srg-ssr.ch 
[I] HttpHeader.h : 192 -  -> Connection: close
[I] HttpHeader.h : 192 -  -> Accept: audio/mp3 
[I] HttpHeader.h : 298 -  -> <CR LF>
[I] HttpRequest.h : 227 - Request written ... waiting for reply
[I] HttpHeader.h : 225 - HttpHeader::read
[W] HttpHeader.h : 232 - Waiting for data...
[I] HttpHeader.h : 162 - HttpHeader::readLine -> HTTP/1.1 200 OK
[I] HttpHeader.h : 162 - HttpHeader::readLine -> Content-Type: audio/mpeg
[I] HttpHeader.h : 162 - HttpHeader::readLine -> Date: Mon, 20 Feb 2023 09:37:58 GMT
[I] HttpHeader.h : 162 - HttpHeader::readLine -> icy-br:128
[I] HttpHeader.h : 162 - HttpHeader::readLine -> ice-audio-info: bitrate=128
[I] HttpHeader.h : 162 - HttpHeader::readLine -> icy-br:128
[I] HttpHeader.h : 162 - HttpHeader::readLine -> icy-description:SwissJazz_MP3@128Kbps
[I] HttpHeader.h : 162 - HttpHeader::readLine -> icy-genre:Entertainment
[I] HttpHeader.h : 162 - HttpHeader::readLine -> icy-mimetype:audio/mpeg
[I] HttpHeader.h : 162 - HttpHeader::readLine -> icy-name:/SwissJazz_Room2
[I] HttpHeader.h : 162 - HttpHeader::readLine -> icy-pub:0
[I] HttpHeader.h : 162 - HttpHeader::readLine -> icy-url:www.srgssr.ch
[I] HttpHeader.h : 162 - HttpHeader::readLine -> Server: Icecast 2.4.0-kh10
[I] HttpHeader.h : 162 - HttpHeader::readLine -> Cache-Control: no-cache, no-store
[I] HttpHeader.h : 162 - HttpHeader::readLine -> Access-Control-Allow-Origin: *
[I] HttpHeader.h : 162 - HttpHeader::readLine -> Access-Control-Allow-Headers: Origin, Accept, X-Requested-With, Content-Type
[I] HttpHeader.h : 162 - HttpHeader::readLine -> Access-Control-Allow-Methods: GET, OPTIONS, HEAD
[I] HttpHeader.h : 162 - HttpHeader::readLine -> Connection: Close
[I] HttpHeader.h : 162 - HttpHeader::readLine -> Expires: Mon, 26 Jul 1997 05:00:00 GMT
[I] HttpHeader.h : 162 - HttpHeader::readLine ->
[W] HttpRequest.h : 153 - no CONTENT_LENGTH found in reply
[I] URLStream.h : 91 - size: 0
[I] AudioCopy.h : 69 - buffer_size=1024
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[E] Fade.h : 271 - setAudioInfo not called
[E] Fade.h : 271 - setAudioInfo not called
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[E] Fade.h : 271 - setAudioInfo not called
[E] Fade.h : 271 - setAudioInfo not called
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[E] Fade.h : 271 - setAudioInfo not called
[E] Fade.h : 271 - setAudioInfo not called
[E] Fade.h : 271 - setAudioInfo not called
[I] AudioCopy.h : 134 - StreamCopy::copy 1024 -> 1024 -> 1024 bytes - in 1 hops
[E] Fade.h : 271 - setAudioInfo not called
[E] Fade.h : 271 - setAudioInfo not called
[E] Fade.h : 271 - setAudioInfo not called
...
...

**Comment by pschatzmann** [comment]
You obviously try to output some audio where the audio format is not known. Why don't you just configure it with a default value ?
The error message is pretty explicit which how to do this...

**Reply by AllanTee** [reply]
The audio is from: "http://stream.srg-ssr.ch/m/rsj/mp3_128".
auto cfg = analog.defaultConfig(TX_MODE);
analog.begin(cfg);
